internal/config/ethereum/ethereum.go

```diff
@@ -16,26 +16,51 @@
 package ethereum
 
 import (
+	"errors"
 	"fmt"
 	"io/ioutil"
+	"net/http"
 	"strings"
 
 	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/makerdao/oracle-suite/internal/rpcsplitter"
 
 	"github.com/makerdao/oracle-suite/pkg/ethereum"
 	"github.com/makerdao/oracle-suite/pkg/ethereum/geth"
 )
 
+const splitterVirtualHost = "makerdao-splitter"
+
 //nolint:unlambda
-var ethClientFactory = func(rpc string) (geth.EthClient, error) {
-	return ethclient.Dial(rpc)
+var ethClientFactory = func(endpoints []string) (geth.EthClient, error) {
+	switch len(endpoints) {
+	case 0:
+		return nil, errors.New("missing address to a RPC client in the configuration file")
+	case 1:
+		return ethclient.Dial(endpoints[0])
+	default:
+		splitter, err := rpcsplitter.NewTransport(endpoints, splitterVirtualHost, nil)
+		if err != nil {
+			return nil, err
+		}
+		rpcClient, err := rpc.DialHTTPWithClient(
+			fmt.Sprintf("http://%s", splitterVirtualHost),
+			&http.Client{Transport: splitter},
+		)
+		if err != nil {
+			return nil, err
+		}
+		return ethclient.NewClient(rpcClient), nil
+	}
 }
 
 type Ethereum struct {
-	From     string `json:"from"`
-	Keystore string `json:"keystore"`
-	Password string `json:"password"`
-	RPC      string `json:"rpc"`
+	From     string      `json:"from"`
+	Keystore string      `json:"keystore"`
+	Password string      `json:"password"`
+	RPC      interface{} `json:"rpc"`
```

Comment:
How about having two separate keys. Otherwise, the migration will be more complicated and we could actually use both `rpc` key and the new one for two different purposes.
The new one could be used for reliable reading of on-chain data - with redundancy, and the old one would be for sending transactions - unless we want to be using multiple nodes for both "call" and "send".

Reply:
I'm not sure if I understand you correctly. In your previous PR (#181) you've already created a separate key for that.

Reply:
you're right